src/overlays/actors/ovl_Bg_Ydan_Sp/z_bg_ydan_sp.c

```diff
@@ -63,29 +73,377 @@ static ColliderTrisInit D_808C09A8 = {
+    Actor_ProcessInitChain(thisx, sInitChain);
+    this->isDestroyedSwitchFlag = thisx->params & 0x3F;
+    this->burnSwitchFlag = (thisx->params >> 6) & 0x3F;
+    thisx->params = (thisx->params >> 0xC) & 0xF;
+    DynaPolyActor_Init((DynaPolyActor*)thisx, DPM_PLAYER);
+    Collider_InitTris(globalCtx, &this->trisCollider);
+    Collider_SetTris(globalCtx, &this->trisCollider, thisx, &sTrisInit, this->trisColliderItems);
+    if (thisx->params == WEB_FLOOR) {
+        CollisionHeader_GetVirtual(&D_06006460, &colHeader);
+        this->actionFunc = &BgYdanSp_FloorWebIdle;
+
+        for (i = 0; i < 3; i++) {
+            tri[i].x = ti0->dim.vtx[i].x + thisx->world.pos.x;
+            tri[i].y = ti0->dim.vtx[i].y + thisx->world.pos.y;
+            tri[i].z = ti0->dim.vtx[i].z + thisx->world.pos.z;
+        }
+
+        Collider_SetTrisVertices(&this->trisCollider, 0, &tri[0], &tri[1], &tri[2]);
+        tri[1].x = tri[0].x;
+        tri[1].z = tri[2].z;
+        Collider_SetTrisVertices(&this->trisCollider, 1, &tri[0], &tri[2], &tri[1]);
+        this->unk16C = 0.0f;
+    } else {
+        CollisionHeader_GetVirtual(&D_06006050, &colHeader);
+        this->actionFunc = &BgYdanSp_WallWebIdle;
+        Actor_SetFocus(thisx, 30.0f);
+        sinsY = Math_SinS(thisx->shape.rot.y);
+        cossY = Math_CosS(thisx->shape.rot.y);
+        nSinsX = -Math_SinS(thisx->shape.rot.x);
+        cossX = Math_CosS(thisx->shape.rot.x);
+
+        for (i = 0; i < 3; i++) {
+            tri[i].x = thisx->world.pos.x + (cossY * ti1->dim.vtx[i].x) - (sinsY * ti1->dim.vtx[i].y * nSinsX);
+            tri[i].y = thisx->world.pos.y + (ti1->dim.vtx[i].y * cossX);
+            tri[i].z = thisx->world.pos.z - (sinsY * ti1->dim.vtx[i].x) + (ti1->dim.vtx[i].y * cossY * nSinsX);
+        }
+
+        Collider_SetTrisVertices(&this->trisCollider, 0, &tri[0], &tri[1], &tri[2]);
+
+        tri[1].x = thisx->world.pos.x + (cossY * ti1->dim.vtx[0].x) - (ti1->dim.vtx[2].y * sinsY * nSinsX);
+        tri[1].y = thisx->world.pos.y + (ti1->dim.vtx[2].y * cossX);
+        tri[1].z = thisx->world.pos.z - (sinsY * ti1->dim.vtx[0].x) + (ti1->dim.vtx[2].y * cossY * nSinsX);
+        Collider_SetTrisVertices(&this->trisCollider, 1, &tri[0], &tri[2], &tri[1]);
+    }
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, thisx, colHeader);
+    this->timer = 0;
+    if (Flags_GetSwitch(globalCtx, this->isDestroyedSwitchFlag)) {
+        Actor_Kill(&this->dyna.actor);
+    }
+}
+
+void BgYdanSp_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgYdanSp* this = THIS;
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyTris(globalCtx, &this->trisCollider);
+}
+
+void* BgYdanSp_UpdateFloorWebCollision(BgYdanSp* this) {
+    s16 newY;
+    CollisionHeader* colHeader;
+
+    colHeader = (CollisionHeader*)SEGMENTED_TO_VIRTUAL(&D_06006460);
+    colHeader->vtxList = (Vec3s*)SEGMENTED_TO_VIRTUAL(colHeader->vtxList);
```

Comment:
We generally don't cast the result of `VIRTUAL` macros (or anything that returns a `void*` really)
```suggestion
    colHeader = SEGMENTED_TO_VIRTUAL(&D_06006460);
    colHeader->vtxList = SEGMENTED_TO_VIRTUAL(colHeader->vtxList);
```